spring-boot-autoconfigure/src/main/java/org/axonframework/springboot/autoconfig/AxonDbSchedulerAutoConfiguration.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2010-2023. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.springboot.autoconfig;
+
+import com.github.kagkarlsson.scheduler.Scheduler;
+import com.github.kagkarlsson.scheduler.task.Task;
+import org.axonframework.common.transaction.TransactionManager;
+import org.axonframework.config.Configuration;
+import org.axonframework.config.ConfigurationScopeAwareProvider;
+import org.axonframework.deadline.DeadlineManager;
+import org.axonframework.deadline.dbscheduler.DbSchedulerBinaryDeadlineDetails;
+import org.axonframework.deadline.dbscheduler.DbSchedulerDeadlineManager;
+import org.axonframework.eventhandling.EventBus;
+import org.axonframework.eventhandling.scheduling.EventScheduler;
+import org.axonframework.eventhandling.scheduling.dbscheduler.DbSchedulerBinaryEventData;
+import org.axonframework.eventhandling.scheduling.dbscheduler.DbSchedulerEventScheduler;
+import org.axonframework.messaging.ScopeAwareProvider;
+import org.axonframework.serialization.Serializer;
+import org.axonframework.tracing.SpanFactory;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.autoconfigure.AutoConfiguration;
+import org.springframework.boot.autoconfigure.AutoConfigureAfter;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.context.annotation.Bean;
+
+/**
+ * Auto configuration class for the deadline manager and event scheduler using db-scheduler.
+ *
+ * @author Gerard Klijs
+ * @since 4.8.0
+ */
+@AutoConfiguration
+@ConditionalOnBean(Scheduler.class)
+@AutoConfigureAfter(value = {AxonServerAutoConfiguration.class, AxonTracingAutoConfiguration.class},
+        name = {"com.github.kagkarlsson.scheduler.boot.autoconfigure.DbSchedulerAutoConfiguration"})
+public class AxonDbSchedulerAutoConfiguration {
+
+    @Bean
+    Task<DbSchedulerBinaryEventData> dbSchedulerBinaryEventDataTask() {
+        return DbSchedulerEventScheduler.binaryTask();
+    }
+
+    @Bean
+    Task<DbSchedulerBinaryDeadlineDetails> dbSchedulerBinaryDeadlineDetailsTask() {
```

Comment:
:thinking:  why can't they both be there? Does the `Scheduler` from db-scheduler not add a list of `Task` instances?

Apart from that, I do agree adding a property-files property to switch between the both is quite redundant.

Reply:
They have the same name.. we could give them different names, but that would complicate the tokens. And they should only have either one.

Reply:
:thinking: I think the bean name is the method name. So, since the names are different, both `Task` beans would exist in the Application Context, right?

Reply:
Yes, both of the  `Task` would exist in the Application Context. The problem is with initializing the `Scheduler`. The task name they use is the same static one from the token, so to also split that out would need a lot of additional if statements.

Reply:
Gotcha, I think I know sufficient.
What about we rename the bean creation methods here to *not* refer to whether they're binary or human-readable.

That way, a user can override these beans by exposing a bean of type `Task`, with the same name, but internally return the human-readable format instead.
By making these beans conditional, I think that'll work.
What do you think?

Reply:
They still need to set the property on the components. We could add a spring property to switch easily, but they also could disable this auto-configuration if they want the human-readable format?

Reply:
Make it easy to overwrite the task beans.